packages/live-share-inking/src/core/InkingManager.ts

```diff
@@ -0,0 +1,1293 @@
+import { Stroke, IStroke, IStrokeCreationOptions, StrokeType } from "./Stroke";
+import { InputFilter, InputFilterCollection } from "../input/InputFilter";
+import { JitterFilter } from "../input/JitterFilter";
+import { generateUniqueId, pointerEventToPoint } from "./Utils";
+import { InputProvider } from "../input/InputProvider";
+import { PointerInputProvider } from "../input/PointerInputProvider";
+import { DefaultHighlighterBrush, DefaultLaserPointerBrush, DefaultPenBrush, IBrush } from "./Brush";
+
+interface CoalescedPointerEvent extends PointerEvent {
+    getCoalescedEvents(): PointerEvent[];
+}
+
+/**
+ * @hidden
+ * Retrieves a list of coalesced PointerEvent objects from a single PointerEvent.
+ * PointerEvent.getCoalescedEvents() is an experimental feature and TypeScript doesn't
+ * yet declare it.
+ * For more info: https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/getCoalescedEvents
+ * @param event The PointerEvent to get coalesced events from.
+ * @returns An array of coalesced PointerEvent objects.
+ */
+function getCoalescedEvents(event: PointerEvent): PointerEvent[] {
+    if ('getCoalescedEvents' in event) {
+        const events: PointerEvent[] = (event as CoalescedPointerEvent).getCoalescedEvents();
+
+        // Firefox can return an empty list. See https://bugzilla.mozilla.org/show_bug.cgi?id=1511231.
+        if (events.length >= 1) {
+            return events;
+        }
+    }
+
+    return [event];
+}
+
+/**
+ * Defines available inking tools.
+ */
+export enum InkingTool {
+    Pen,
+    LaserPointer,
+    Highlighter,
+    Eraser,
+    PointEraser
+}
+
+type StrokeBasedTool = InkingTool.Pen | InkingTool.LaserPointer | InkingTool.Highlighter;
+
+/**
+ * The event emitted by InkingManager when the canvas is cleared.
+ */
+export const ClearEvent: symbol = Symbol();
+/**
+ * The event emitted by InkingManager when a stroked is added.
+ */
+export const StrokesAddedEvent: symbol = Symbol();
+/**
+ * The event emitted by InkingManager when a stroked is removed.
+ */
+export const StrokesRemovedEvent: symbol = Symbol();
+
+/**
+ * Defines the arguments of the PointerMovedEvent.
+ */
+export interface IPointerMovedEventArgs {
+    position?: IPoint;
+}
+
+/**
+ * The event emitted by InkingManager when the pointer moves over the canvas.
+ */
+export const PointerMovedEvent: symbol = Symbol();
+
+/**
+ * Defines the arguments of the BeginStrokeEvent.
+ */
+export interface IBeginStrokeEventArgs {
+    /**
+     * The id of the new stroke.
+     */
+    strokeId: string;
+    /**
+     * The type of the new stroke.
+     */
+    type: StrokeType;
+    /**
+     * The brush of the new stroke.
+     */
+    brush: IBrush;
+    /**
+     * The starting point of the new stroke.
+     */
+    startPoint: IPointerPoint;
+}
+
+/**
+ * The event emitted by InkingManager when a stroke begins.
+ */
+export const BeginStrokeEvent: symbol = Symbol();
+
+export enum StrokeEndState {
+    Ended,
```

Comment:
Another note about enum values... If they're going to be transmitted over the wire you should probably use string enums so `ended = "ended",` for better readability of the transmitted data. Otherwise you're values will be 0, 1, etc.

Reply:
I've deliberately not done that so as to minimize the amount of data over the write. I can do it for this one because it's only transmitted via events, but enum values that are stored in the SharedMap, I actually want to use as little space as possible and think we should stick with numeric values even if that means it's not human readable.

Reply:
As per separate discussion, keeping non-string enums that help reduce BoW.